src/containers/AssetForm.js

```diff
@@ -1,16 +1,416 @@
+  borderStyle: 'solid',
+  borderColor: '#d1d1d1',
+  padding: '10px 20px'
+};
+
+const storageFormatGroupStyle = {
+  display: 'flex',
+  borderLeft: '1px solid #d1d1d1'
+};
+
+const storageFormatButtonStyle = {
+  width: 'auto',
+  borderWidth: '1px 1px 1px 0',
+  borderStyle: 'solid',
+  borderColor: '#d1d1d1',
+  padding: "10px 20px"
+};
 
 /*
   Renders the form for the creation/editing of an Asset.
   */
-const AssetForm = ({ match }) => (
-  <div className="App-main">
-    <AssetFormHeader title={match.url === '/asset/create' ? 'Create new asset' : 'Editing: Some asset'} />
-    <div>
-      <h1>Asset Creation/Editing Form</h1>
-    </div>
-  </div>
+class AssetForm extends Component {
+
+  constructor() {
+    super();
+
+    this.fetchMatchingUsers = _.debounce(this.fetchMatchingUsers, 200);
+    this.handleChange = this.handleChange.bind(this);
+
+    this.state = {
+      // asset state
+      name: null,
+      department: null,
+      purpose: null,
+      research: null,
+      owner: null,
+      private: null,
+      personal_data: null,
+      data_subject: [],
+      data_category: [],
+      recipients_category: null,
+      recipients_outside_eea: null,
+      retention: null,
+      retention_other: null,
+      risk_type: [],
+      storage_location: null,
+      storage_format: [],
+      paper_storage_security: [],
+      digital_storage_security: [],
+
+      // non-asset state
+
+      // the list of matching users returned from lookup
+      matchingUsers: [],
+      // the selected owner's full name
+      ownerName: ""
+    }
+  }
+
+  /*
+  Wrapper for fetch() that handles errors / unmarshalls JSON
+   */
+  fetch(url, options, cb) {
+    fetch(url, options).then(response => {
+      if (response.ok) {
+        return response.json()
+      } else {
+        this.props.handleMessage('Network Error: ' + response.statusText)
+      }
+    }).then(data => data && cb(data)).catch(
+        error => this.props.handleMessage('Network Error: ' + error)
+    );
+  }
+
+  /*
+  If the form is in edit mode - fetch the asset (and the owner's name - if there is one)
+   */
+  componentDidMount() {
+    if (this.props.match.url !== '/asset/create') {
+      this.fetch(config.ENDPOINT_ASSETS + this.props.match.params.asset + '/', {}, data => {
+        this.setState(data);
+        if (this.state.owner) {
+          this.fetchOwnerName();
+        }
+      });
+    }
+  }
+
+  /*
+  Fetch the owners name for the lookup API.
+   */
+  fetchOwnerName() {
+    this.fetch(ENDPOINT_PEOPLE + this.state.owner, {
+      headers: new Headers({
+        'Authorization': 'Bearer ' + ACCESS_TOKEN
+      })
+    }, data => {
+      this.setState({ownerName: data.visibleName})
+    });
+  }
+
+  /*
+  As the user types into the owner AutoComplete the ownerName is updated and a list of users
+  matching the searchText is fetched.
+  Deleting all text is detected and the owner field is cleared to allow the deleting of an owner.
+   */
```

Comment:
In the fullness of time we should re-factor this into a separate component. For this story, however, this can stay where it is.